### DISCUSSION TITLE
v3 livewire.js Not Found on production server

### DISCUSSION THREAD

**Discussion by faizananwerali** [livewire/livewire#6573] [1. Help] [answered]
`GET https://domain.com/livewire/livewire.js?id=11c49d7e net::ERR_ABORTED 404 (Not Found)`

In Livewire v2, I could use this `php artisan livewire:publish --assets` to publish assets, not anymore.

and in composer.json

```json
        "post-update-cmd": [
            "@php artisan vendor:publish --tag=laravel-assets --ansi --force",
            "@php artisan vendor:publish --tag=livewire:assets --ansi --force",
            "@php artisan filament:upgrade"
        ],
```

but this is not working anymore, all the work I've done past two weeks is inaccessible because livewire.js is not found.

Can anybody please help me here

**Comment by tahirafridi** [comment, marked as answer]
@faizananwerali here is the solution for your problem https://github.com/livewire/livewire/discussions/6328

**Comment by andhikapepe** [comment]
- I just copy inside all dist from _vendor/livewire/livewire/dist/*_ to inside folder _public/_ and then rename folder _dist_ to _livewire_, my structure becomes _public/livewire/*_
- inside routes\web.php I just add
```
use Livewire\Livewire;
Livewire::setScriptRoute(function ($handle) {
    $urlPath = parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH);
    $segments = explode('/', trim($urlPath, '/'));
    // Use the null coalescing operator to simplify the conditional assignment
    $rootFolder = count($segments) >= 2 ? $segments[0] : 'localhost';
    return Route::get($rootFolder . '/public/livewire/livewire.js', $handle);
});
```
hopefully can help.

Note: **this is just one way that can be used to make things easier**
